src/Meilisearch/Index.cs

```diff
@@ -67,23 +79,49 @@ public async Task<bool> Delete()
         /// Add documents.
         /// </summary>
         /// <param name="documents">Documents to add.</param>
+        /// <param name="primaryKey">Primary key for the documents.</param>
         /// <typeparam name="T">Type of the document. Even though documents are schemaless in MeiliSearch, making it typed helps in compile time.</typeparam>
         /// <returns>Returns the updateID of this async operation.</returns>
-        public async Task<UpdateStatus> AddDocuments<T>(IEnumerable<T> documents)
+        public async Task<UpdateStatus> AddDocuments<T>(IEnumerable<T> documents, string primaryKey = default)
         {
-            var responseMessage = await this.client.PostAsJsonAsync($"/indexes/{this.Uid}/documents", documents);
+            HttpResponseMessage responseMessage;
+            string uri = $"/indexes/{this.Uid}/documents";
+            if (primaryKey == default)
+            {
+                responseMessage = await this.client.PostAsJsonAsync(uri, documents);
+            }
+            else
+            {
+                this.PrimaryKey = primaryKey;
+                uri = QueryHelpers.AddQueryString(uri, new { primaryKey = primaryKey }.AsDictionary());
+                responseMessage = await this.client.PostAsJsonAsync(uri, documents);
+            }
+
             return await responseMessage.Content.ReadFromJsonAsync<UpdateStatus>();
         }
 
         /// <summary>
         /// Update documents.
         /// </summary>
         /// <param name="documents">Documents to update.</param>
+        /// <param name="primaryKey">Primary key for the documents.</param>
         /// <typeparam name="T">Type of document. Even though documents are schemaless in MeiliSearch, making it typed helps in compile time.</typeparam>
         /// <returns>Returns the updateID of this async operation.</returns>
-        public async Task<UpdateStatus> UpdateDocuments<T>(IEnumerable<T> documents)
+        public async Task<UpdateStatus> UpdateDocuments<T>(IEnumerable<T> documents, string primaryKey = default)
         {
-            var responseMessage = await this.client.PutAsJsonAsync($"/indexes/{this.Uid}/documents", documents);
+            HttpResponseMessage responseMessage;
+            string uri = $"/indexes/{this.Uid}/documents";
+            if (primaryKey == default)
+            {
+                responseMessage = await this.client.PutAsJsonAsync(uri, documents);
+            }
+            else
+            {
+                this.PrimaryKey = primaryKey;
```

Comment:
same here: don't update the primary key of the index instance

Reply:
Done